packages/odata-common/src/request-builder/action-function-import-request-builder-base.ts

```diff
@@ -10,7 +12,9 @@ import { MethodRequestBuilder } from './request-builder-base';
 export abstract class ActionFunctionImportRequestBuilderBase<
   ReturnT,
   RequestConfigT extends ODataRequestConfig
-> extends MethodRequestBuilder<RequestConfigT> {
+> extends MethodRequestBuilder<RequestConfigT>  implements WithBatchReference{
+
+  private _batchReference: BatchReference = { id: uuid() };
```

Comment:
true, I wanted to keep the implementation of the get as short as possible to avoid code duplication. I had in each method then `if(!batchReference){this.setBatchReference(uuid())}` which I could not put in a helper method. Hence I decided to initialize on creation, which is I think acceptable because it is really a small object.

Reply:
was a leftover - removed

Reply:
Understood.
The cost is about the number of calls to `uuid`, which is related to performance and memory.
I guess for normal cases, it might be fine.
For big number of big entities, it might be a problem, which happened before when the memory issue was raised.

I tried the following:
- The [mixin](https://codeburst.io/multiple-inheritance-with-typescript-mixins-d92d01198907) does partial work, and the methods of the parent class cannot be reused.
- This package [ts-mixer](https://www.npmjs.com/package/ts-mixer) seems to be helpful

Reply:
Here is the playground [test](https://www.typescriptlang.org/play?target=5&ts=4.1.5#code/MYGwhgzhAEBKCmBHArvCAXAQsgliAJvAE4DCA9gLYVkB2A3gLABQ008AHvMMuvABQBKRi2gBfZuKbNQkGAHUc6ABaYw6YEoQAzYawD6ReFoD8ALmgYiOGgHNoAXmjIahLdfj4A3M1Y346bUFdVmgcLT4AQmUcCAA6AyMhHxDWaLiErQdoAHJkXHxs5NZJYolmZmteIi0wYHg4JFQMTDxCUkpqGgVlVXVNIzZ2XhcYBBQ0LFwCYnIqWmEyqSYAemXoChx2a2h8MjRoGjJ0aAB3MiIAa2ZV6BkoaBJDNXgxpsnW4lCKAAcQeAp4DR0KNGhMWtN2nMuooVGoNNoFis1pJrmtkBB6sCALQbThEaBac6DYb4ax2CjIEDoPD1O4YiDScD3AAi8D+vFeYKmbWJgPw8hhvXhRkRKKYwFoGGgx0cNHgJ2grPZL1BzW5xEE3iRThcRnc+GkkrIf1iIDINj46Hihi0Ai1Vr8ASMmtR0AAbmAQKhQjRFDhPTgAF4eQ00CDG+Cm82W62JTxAA)

Reply:
ok then I make it lazy if we have the experience that there are many request build und UUID is called often.